src/authInfo.ts

```diff
@@ -1096,30 +1111,33 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     };
   }
 
-  private async retrieveUsername(
-    instanceUrl: string,
-    accessToken: string,
-    orgId: string,
-    userId: string
-  ): Promise<Optional<string>> {
+  private async retrieveUserInfo(instanceUrl: string, accessToken: string): Promise<Optional<UserInfo>> {
     // Make a REST call for the username directly.  Normally this is done via a connection
     // but we don't want to create circular dependencies or lots of snowflakes
     // within this file to support it.
-    const apiVersion = 'v42.0'; // hardcoding to v42.0 just for this call is okay.
+    const apiVersion = 'v51.0'; // hardcoding to v51.0 just for this call is okay.
```

Comment:
suggestion: that userinfo response contains a urls objects with versions in it.  we could not hardcode at all in this new approach

https://help.salesforce.com/articleView?id=sf.remoteaccess_using_userinfo_endpoint.htm&type=5

Reply:
it gives you back the url for the /sobjects call

<sobjects>https://yourInstance.salesforce.com/services/data/v{version}/sobjects/</sobjects>

Reply:
💯

Reply:
I'll look at that possibility

Reply:
@amphro  grabbing the url still requires inserting a version number.